src/Http/Request.php

```diff
@@ -176,8 +177,34 @@ public function getCookies()
 		return $this->cookies;
 	}
 
-
-	/********************* method & headers ****************d*g**/
+    /**
+     * Provides a flexible generic get method.
+     * It follows an order of precedence: GET, POST, FILE.
+     *
+     * @param $key
+     * @param $default
+     *
+     * @return mixed
+     */
+    public function get($key, $default = NULL)
```

Comment:
I think it worth it. 
This central method (is like a shortcut), the user won't be afraid about the HTTP verbs used in the Request, and could easily allow interface for other libraries.